### PR TITLE
bcompare: add darwin support

### PR DESCRIPTION
**Pull request by arkivm**
###### Motivation for this change
bcompare: Add support for darwin platforms

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ktor — APPROVED** [review]
OK

**Review by ktor — APPROVED** [review]

**Comment by arkivm** [comment]
@SuperSandro2000 poke

**Review by ktor — APPROVED** [review]

File: pkgs/applications/version-management/bcompare/default.nix [hunk 1]
```diff
@@ -1,54 +1,26 @@
...
-    gdk-pixbuf
-    bzip2
-  ];
-
-  dontBuild = true;
-  dontConfigure = true;
-  dontWrapQtApps = true;
+  src = srcs.${stdenv.hostPlatform.system} or throwSystem;
 
   meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/bcompare/default.nix:25]: meta should always be last.

File: pkgs/applications/version-management/bcompare/default.nix [hunk 2]
```diff
@@ -1,54 +1,26 @@
-{ autoPatchelfHook, bzip2, cairo, coreutils, fetchurl, gdk-pixbuf, pango, gtk2, kcoreaddons, ki18n, kio, kservice, lib, qt4, qtbase, stdenv, runtimeShell }:
+{ autoPatchelfHook, bzip2, cairo, coreutils, fetchurl, gdk-pixbuf, glibc, pango, gtk2, kcoreaddons, ki18n, kio, kservice, lib, qt4, qtbase, stdenv, runtimeShell, unzip }:
```
Comment (SuperSandro2000) [inline pkgs/applications/version-management/bcompare/default.nix:1]:
```suggestion
{ lib, autoPatchelfHook, bzip2, cairo, coreutils, fetchurl, gdk-pixbuf, glibc, pango, gtk2, kcoreaddons, ki18n, kio, kservice,
, qt4, qtbase, stdenv, runtimeShell, unzip
}:
```

Can we switch to qt5?

Comment (ktor) [reply pkgs/applications/version-management/bcompare/default.nix:1]: Just tested it to work without **qt4** and **qtbase** dependencies. Can you please confirm on **darwin** @arkivm ?

Comment (arkivm) [reply pkgs/applications/version-management/bcompare/default.nix]: Thanks @ktor. Works on aarch64-darwin.

**Comment by arkivm** [comment]
on darwin, `otool -L ./result/Applications/BCompare.app/Contents/MacOS/BCompare` gives this 

```
...
	/usr/lib/libbz2.1.0.dylib (compatibility version 1.0.0, current version 1.0.5)
	/usr/lib/libresolv.9.dylib (compatibility version 1.0.0, current version 1.0.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
```
Are these libs allowed to be impure or should we fix it with `install_tool_helper`?

**Comment by arkivm** [comment]
@SuperSandro2000 @ktor any thoughts on the macos libraries requirement?

**Comment by ktor** [comment]
> @SuperSandro2000 @ktor any thoughts on the macos libraries requirement?

Hi @arkivm, unfortunately, I've never solved similar problem. I am curious about your solution though.

**Review by ktor — APPROVED** [review]

**Pull request merged**